### DISCUSSION TITLE
Error on task  Git based WIKI Folder Updater from Black Marble - Version: 2.0.5

### DISCUSSION THREAD

**Discussion by GuilherrmeLima** [rfennell/AzurePipelines#1446] [General]
When using the task in the latest version available, I get this error starting the task.

(I'm using Azure Devops Server 2020, build agent in the latest version and windows server 2012 operating system)

``` bash
Starting: Git based WIKI Folder Updater
==============================================================================
Task         : Git based WIKI Folder Updater from Black Marble
Description  : A tools to update or create a set of files in a folder in WIKI based in a Git repo 
Version      : 2.0.5
Author       : Richard Fennell
Help         : Version: 2.0.5. [More Information](https://github.com/rfennell/vNextBuild/wiki/Wiki-Updater-Task/)
==============================================================================
##[error]Unhandled: Invalid regular expression flags
##[error]C:\azagents\azagent\_work\_tasks\WikiFolderUpdaterTask_2f22eed4-2ef7-407e-98d8-3b20ca0a5352\2.0.5\node_modules\simple-git\dist\cjs\index.js:3170
      new LineParser(/^([*+]\s)?(\S+)\s+([a-z0-9]+)\s?(.*)$/s, (result, [current, name, commit, label]) => {
                     ^

SyntaxError: Invalid regular expression flags
    at createScript (vm.js:56:10)
    at Object.runInThisContext (vm.js:97:10)
    at Module._compile (module.js:549:28)
    at Object.Module._extensions..js (module.js:586:10)
    at Module.load (module.js:494:32)
    at tryModuleLoad (module.js:453:12)
    at Function.Module._load (module.js:445:3)
    at Module.require (module.js:504:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (C:\azagents\azagent\_work\_tasks\WikiFolderUpdaterTask_2f22eed4-2ef7-407e-98d8-3b20ca0a5352\2.0.5\GitWikiFuntions.js:13:22)
Finishing: Git based WIKI Folder Updater
```

**Comment by rfennell** [comment]
There is no feature change between the last 1.x version of this task and 2.x. All I did was update all the NPM packages and the Node Runner for Node 10 to 16.

For now I would revert to using the V1 of the task.

I will do some testing on 2020 server and see if I can repo the problem

@GuilherrmeLima  can you confirm it was working with V1, and reverting fixes your issues?

**Reply by rfennell** [reply]
@GuilherrmeLima can I check, do you have Node16 installed on your build agent?

Also, what happens if you force the pipeline to use Node10?

> To start a task using node 10, we can set the pipeline variable AGENT_USE_NODE10 to true.

[See the task version migration notes for details](https://github.com/microsoft/azure-pipelines-tasks/blob/master/docs/migrateNode16.md#testing-the-changes)

**Reply by GuilherrmeLima** [reply]
that's right, exactly on that agent I didn't have node 16 installed, I installed it and the problem was solved.
Thank you for the information and quicker in solving the case.

**Reply by rfennell** [reply]
@GuilherrmeLima Great that fixed it,  would you mind marking this thread as answer so other can find it easily. 

I fear that this might be an issue for any tasks that are widely used that OSS move over Node16 runners. It is good that the migration notes advise to leave the option for Node10 as a fall back (if the variable is set), but people might miss that, an answer flag here might help
